### ISSUE TITLE
WebP-only archive thumbnail generation

### ISSUE THREAD

**Issue by RePod** [Difegue/LANraragi#238]
Stable and Nightly, WSL on 1903

TLDR is Image::Scale doesn't take WebP.

### Replication

To replicate the problem, use this test archive: [webp test.zip](https://github.com/Difegue/LANraragi/files/3937050/webp.test.zip)
 - With the PNG, there will be no issues as [WebP is filtered out and the PNG remains](https://github.com/Difegue/LANraragi/blob/9359b7b2c6bd687480b6f602bec87d77d89f25c1/lib/LANraragi/Utils/Archive.pm#L110) for thumbnail generation.
 - Without the PNG and without updating the regex, the array is empty and that becomes a problem like you would expect:
   - *Use of uninitialized value in subroutine entry at /usr/local/lib/perl5/site_perl/Archive/Peek/Libarchive.pm line 13/20.*
 - Without the PNG and updating the regex to account for the WebP, the issue is now in [generate_thumbnail](https://github.com/Difegue/LANraragi/blob/9359b7b2c6bd687480b6f602bec87d77d89f25c1/lib/LANraragi/Utils/Archive.pm#L30) as Image::Scale does not seem to support [WebP as an input](https://metacpan.org/pod/Image::Scale#DESCRIPTION). Thus execution dies and claims the image is invalid (which is true to Image::Scale, but that's why we're here).

### Affected areas

Primarily the Index/Archive list. The entry is displayed with the generic thumbnail, having failed, but once accessed dies as stated above. If Reader can be reached (in any of the above scenarios), WebP support is placed on the client so that's a non-issue.

Reader [proactively considers WebP](https://github.com/Difegue/LANraragi/blob/24819d4ac5e00f4922638e157eaf4f8f05801a14/lib/LANraragi/Utils/Generic.pm#L30).

### Solutions considered

 - Use the workaround of having a filler image. In a perfect world this would be a cover.
   - But that's effort!
 - Abandon WebP-only archives.
 - [Image::Magick](https://metacpan.org/pod/Image::Magick)
   - I don't know the complete history, but there's unused leftovers in Reader for [overview thumbnails](https://github.com/Difegue/LANraragi/blob/dev/lib/LANraragi/Model/Reader.pm#L30). As a side effect of being unused, said overview thumbnails are the original images themselves.
   - It might be worth considering offloading reliance to imagemagick to future proof support for situations like this.
 - [Imager::File::WEBP](https://metacpan.org/pod/Imager::File::WEBP)
   - Just something else for reading WebP.

**Comment by Difegue** [comment]
Thank you for the detailed report! I'll have to write a bit of history here:

Initially the module used for resizing images was indeed Image::Magick; I replaced it with Image::Scale as I wanted something a bit simpler and faster than the full Magick suite.  

The leftovers in the Reader code are from an old feature where images were resized on-the-fly with Magick before outputting them to the client to save on bandwidth. Since I dropped Magick I removed the feature but left the code in just in case.  

Image::Scale works for what it needs to do but it's a bit limited and requires a bunch of crap to compile during install. (one lib per image format basically)

I've been thinking about putting Image::Magick back in for a while but never got to it since Scale does work well enough. WebP thumbnails are a good enough reason to do it!  
The call in [Archive.pm](https://github.com/Difegue/LANraragi/blob/9359b7b2c6bd687480b6f602bec87d77d89f25c1/lib/LANraragi/Utils/Archive.pm#L110) should also be refactored to use is_image from the Generic functions.

**Comment by Difegue** [comment]
This last commit should now make webp thumbnails work on all platforms. 🎉  
Alpine ships with magick v7 and debian/ubuntu still uses v6, but both should work the same as they rely on google's libwebp tools.
( sourced from http://www.imagemagick.org/discourse-server/viewtopic.php?t=34785 )
